lib/chai/core/assertions.js

```diff
@@ -536,6 +536,13 @@ module.exports = function (chai, _) {
       case 'string':
         itemsCount = val.length;
         break;
+      case 'map':
+      case 'set':
+        itemsCount = val.size;
+        break;
+      case 'weakmap':
+      case 'weakset':
+        throw new TypeError('.empty was called on a weak collection');
```

Comment:
I think that to keep consistency with the TypeError messages added to #812, this should be something like this:
`.empty was passed a weak collection`